WcaOnRails/app/models/registration.rb

```diff
@@ -24,12 +27,30 @@ class Registration < ActiveRecord::Base
     end
   end
 
-  def pending?
-    accepted_at.nil?
+  def deleted?
+    !deleted_at.nil?
   end
 
   def accepted?
-    !pending?
+    !deleted? && !accepted_at.nil?
+  end
+
+  def pending?
+    !accepted? && !deleted?
+  end
+
+  def checked_status
```

Comment:
Ahhh, sorry, I totally forgot how this is supposed to work. Please ignore this commet. It's okay for a registration to be both accepted and deleted, that means that it's deleted =)